lib/elixir/test/elixir/module/types/descr_test.exs

```diff
@@ -212,6 +371,39 @@ defmodule Module.Types.DescrTest do
 
       assert union(atom([:foo, :bar]), dynamic()) |> to_quoted_string() ==
                "dynamic() or (:bar or :foo)"
+
+      assert intersection(dynamic(), map(a: integer())) |> to_quoted_string() ==
+               "dynamic() and %{:a => integer()}"
+    end
+
+    test "map" do
+      assert map() |> to_quoted_string() == "%{...}"
+      assert map(a: integer()) |> to_quoted_string() == "%{:a => integer()}"
+      assert map([a: float()], :open) |> to_quoted_string() == "%{..., :a => float()}"
+
+      # TODO: support this simplification
+      # assert difference(map(), map([a: term()], :open)) |> to_quoted_string() ==
+      #          "%{..., :a => not_set()}"
+
+      assert map(a: integer(), b: atom()) |> to_quoted_string() ==
+               "%{:a => integer(), :b => atom()}"
```

Comment:
This test is flaky due to non-deterministic key ordering, I think we could sort them in to_quoted?